jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketConstants.java

```diff
@@ -8,14 +8,54 @@
  * SPDX-License-Identifier: EPL-2.0 
  * 
  * Contributors: 
- *     Giancarlo Pernudi Segura - initial API and implementation 
+ *     Giancarlo Pernudi Segura - initial API and implementation
+ *     Lidia Ataupillco Ramos
  *******************************************************************************/
 
 package org.eclipse.lsp4jakarta.jdt.core.websocket;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
 import org.eclipse.lsp4j.DiagnosticSeverity;
 
 public class WebSocketConstants {
     public static final String DIAGNOSTIC_SOURCE = "jakarta-websocket";
-    public static final DiagnosticSeverity SEVERITY = DiagnosticSeverity.Error;
+
+    public static final DiagnosticSeverity ERROR = DiagnosticSeverity.Error;
+    public static final DiagnosticSeverity WARNING = DiagnosticSeverity.Warning;
+
+    public static final String DIAGNOSTIC_ERR_MSG = "Cannot calculate WebSocket diagnostics";
+
+    public static final String DIAGNOSTIC_PATH_PARAMS_ANNOT_MISSING = "Parameter is missing annotation @PathParams.";
+    public static final String DIAGNOSTIC_CODE_PATH_PARMS_ANNOT = "AddPathParamsAnnotation";
+    
+    public static final String DIAGNOSTIC_ON_OPEN_INVALID_PARAMS = "Invalid parameter type. Parameter must be of type Session, EndpointConfig parameter, or if annotated with @PathParam of type String, any Java primitive type or boxed version thereof.";
```

Comment:
Let's use full class name in this message, ie. "must be of type jakarta.websocket.Session, ..". The last half of the sentence feels a little clunky. If I am understanding the flow of when each diagnostic shows up, this diagnostic will not show up if on a String or Java primitive type parameter correct? Then maybe we can omit the last piece with `@PathParam` altogether. What do you think?

Reply:
Yes, you're right. I just pushed a commit with the changes suggested. Thank you!

Reply:
I'm not sure about omitting the @PathParam part as this is how the specification has it but agree on giving out the full name for the classes.

Reply:
Even though that is how it is specified in the spec, if the diagnostic will never show up on a String or Java primitive type parameter (or boxed version thereof) then it is not relevant to include in the message.

Reply:
Yes, that diagnostic message only applies to when the parameter types are checked. If they're not the right ones, that error message is thrown.

Reply:
My understanding of this is that this diagnostic will not show up on any parameter of type String or Java primitive type.
For example, it would not show up on the following (instead the diagnostic would be the one about missing `@PathParam`
```
 public void OnOpen(String stringParam) throws IOException {
```
but would show up on the following:
```
 public void OnOpen(Object objectParam) throws IOException {
```

@dalidia can you please clarify?

Reply:
Ah but in that case you want to indicate to the user all the possible valid types...
How about keeping it as-is, but I wonder if we could display the valid types as a list for readability.
```
    public static final String DIAGNOSTIC_ON_OPEN_INVALID_PARAMS = "Invalid parameter type. Parameter must be of type:
    - jakarta.websocket.Session
    - jakarta.websocket.EndpointConfig
    - annotated with @PathParam and of type String or any Java primitive type or boxed version thereof.";
```
We use a list on a diagnostic message here: https://github.com/eclipse/lsp4jakarta/blob/a195a915af4b7f33d884799927fd786546cd0766/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/beanvalidation/BeanValidationDiagnosticsCollector.java#L194-L198
What do you think?

Reply:
Yes, this diagnostic will not show up for any parameter of type String or Java primitive type, and it will show up in the example provided by Kathryn. The reason why I focused on checking the parameter types first rather than the annotation is because I consider this error more critical from a user's standpoint as this will require some logic change, whereas adding `@PathParam` is a simple fix.

Reply:
Makes sense, thanks @dalidia. In that case I agree with @giancarlopernudisegura that we should keep the `@PathParam` piece in that diagnostic message. What do you think of displaying the diagnostic with a list of supported parameter types.

Reply:
Sure, I'll do that! Thanks Kathryn and Giancarlo!